src/auth/models.hpp

```diff
@@ -302,7 +382,8 @@ class Role {
   Role(const std::string &rolename, const Permissions &permissions);
 #ifdef MG_ENTERPRISE
   Role(const std::string &rolename, const Permissions &permissions,
-       FineGrainedAccessHandler fine_grained_access_handler, Databases db_access = {});
+       FineGrainedAccessHandler fine_grained_access_handler, Databases db_access = {},
+       std::optional<UserImpersonation> usr_imp = {});
```

Comment:
does here {} refer to std::nullopt or default constructed UserImpersonation. Maybe explicitly to std::nullopt makes more sense.